### PR TITLE
fix: goreleaser v2 compat, trigger release-binaries with workflow_run

### PR DESCRIPTION
**Pull request by rvagg**
Ref: https://github.com/ipld/go-car/issues/536

(a small amount of 🤞 going on here, we'll see, this works for lassie)

### CODE REVIEW

File: .github/workflows/release-binaries.yml [hunk 1]
```diff
@@ -2,7 +2,10 @@ name: GoReleaser
 on:
   release:
     types: [ published ]
-
+  workflow_dispatch:
+  workflow_run:
+    workflows: [Releaser]
```
Comment (masih) [inline .github/workflows/release-binaries.yml:7]: Relases created by GitHub actions will not create other workflows, unless they use a GitHub Token other than the built-in default. 

An alternative solution is to delete and recreate the release manually.

I [pointed this out](https://filecoinproject.slack.com/archives/C03KLC57LKB/p1693986844074549) to IPDX team a while back and I believe there has been fixes rolled out to the unified CI? It's worth checking with them.

**Review by masih — APPROVED** [review]
Approved so that you can try the changes here if you wish. We can revert if this doesn't fix it.

File: .github/workflows/release-binaries.yml (hunk 1 above)
Comment (rvagg) [reply .github/workflows/release-binaries.yml:7]: 🤷 this seems to work in the lassie repo where we have essentially the same setup and no custom token configured

Comment (rvagg) [reply .github/workflows/release-binaries.yml:7]: https://github.com/filecoin-project/lassie/actions/runs/9691067118

> Triggered via workflow run 3 months ago

Comment (masih) [reply .github/workflows/release-binaries.yml:7]: Oh nice

**Pull request merged**